tests/ct_connection.c

```diff
@@ -174,6 +174,81 @@ void test_multicluster() {
     redisClusterFree(cc2);
 }
 
+/* Connect to a non-routable address which results in a connection timeout. */
+void test_connect_timeout() {
+    struct timeval timeout = {0, 200000};
+
+    redisClusterContext *cc = redisClusterContextInit();
+    assert(cc);
+
+    /* Configure a non-routable IP address and a timeout */
+    redisClusterSetOptionAddNodes(cc, "192.168.0.0:7000");
+    redisClusterSetOptionConnectTimeout(cc, timeout);
+
+    int status = redisClusterConnect2(cc);
+    assert(status == REDIS_ERR);
+    assert(cc->err == REDIS_ERR_IO);
+    assert(strncmp(cc->errstr, "Connection timed out", 20) == 0);
```

Comment:
Can we avoid the hardcoded `20`? E.g.

    char *str = "Connection timed out";
    assert(strncmp(cc->errstr, str, strlen(str)) == 0);

.... or maybe better: Can't we just use `strcmp`? If errstr is null-terminated, it shouldn't be a problem, should it?